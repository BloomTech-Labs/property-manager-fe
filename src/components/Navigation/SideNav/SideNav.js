import React from 'react';
import clsx from 'clsx';
import { makeStyles, useTheme } from '@material-ui/core/styles';
import Drawer from '@material-ui/core/Drawer';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import List from '@material-ui/core/List';
import CssBaseline from '@material-ui/core/CssBaseline';
import Divider from '@material-ui/core/Divider';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';
import ChevronLeftIcon from '@material-ui/icons/ChevronLeft';
import ChevronRightIcon from '@material-ui/icons/ChevronRight';
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import DashboardIcon from '@material-ui/icons/Dashboard';
import HomeWorkIcon from '@material-ui/icons/HomeWork';
import AccountCircleIcon from '@material-ui/icons/AccountCircle';
import { navigate } from '@reach/router';

const drawerWidth = 240;

const useStyles = makeStyles(theme => ({
  root: {
    display: 'flex'
  },
  appBar: {
    zIndex: theme.zIndex.drawer + 1,
    transition: theme.transitions.create(['width', 'margin'], {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.leavingScreen
    })
  },
  appBarShift: {
    marginLeft: drawerWidth,
    width: `calc(100% - ${drawerWidth}px)`,
    transition: theme.transitions.create(['width', 'margin'], {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.enteringScreen
    })
  },
  menuButton: {
    marginRight: 36
  },
  hide: {
    display: 'none'
  },
  drawer: {
    width: drawerWidth,
    flexShrink: 0,
    whiteSpace: 'nowrap'
  },
  drawerOpen: {
    width: drawerWidth,
    transition: theme.transitions.create('width', {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.enteringScreen
    })
  },
  drawerClose: {
    transition: theme.transitions.create('width', {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.leavingScreen
    }),
    overflowX: 'hidden',
    width: theme.spacing(7) + 1,
    [theme.breakpoints.up('sm')]: {
      width: theme.spacing(9) + 1
    }
  },
  toolbar: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'flex-end',
    padding: theme.spacing(0, 1),
    ...theme.mixins.toolbar
  }
}));

function SideNav() {
  const classes = useStyles();
  const theme = useTheme();
  const [open, setOpen] = React.useState(true);

  const handleDrawerOpen = () => {
    setOpen(true);
  };

  const handleDrawerClose = () => {
    setOpen(false);
  };

  return (
    <>
      <CssBaseline />
      <AppBar
        position="fixed"
        className={clsx(classes.appBar, {
          [classes.appBarShift]: open
        })}
      >
        <Toolbar>
          <IconButton
            color="inherit"
            aria-label="open drawer"
            onClick={handleDrawerOpen}
            edge="start"
          >
            <MenuIcon />
          </IconButton>
          <h4>PropMan</h4>
        </Toolbar>
      </AppBar>
      <Drawer
        variant="permanent"
        className={clsx(classes.drawer, {
          [classes.drawerOpen]: open,
          [classes.drawerClose]: !open
        })}
        classes={{
          paper: clsx({
            [classes.drawerOpen]: open,
            [classes.drawerClose]: !open
          })
        }}
      >
        <div className={classes.toolbar}>
          <IconButton onClick={handleDrawerClose}>
            {theme.direction === 'rtl' ? (
              <ChevronRightIcon />
            ) : (
              <ChevronLeftIcon />
            )}
          </IconButton>
        </div>
        <Divider />
        <List>
          <ListItem button onClick={() => navigate('/dashboard')}>
            <ListItemIcon>
              <DashboardIcon />
            </ListItemIcon>
            <ListItemText primary="Overview" />
          </ListItem>
          <ListItem button onClick={() => navigate('/dashboard/properties')}>
            <ListItemIcon>
              <HomeWorkIcon />
            </ListItemIcon>
            <ListItemText primary="Properties" />
          </ListItem>
          <ListItem button onClick={() => navigate('/dashboard/profile')}>
            <ListItemIcon>
              <AccountCircleIcon />
            </ListItemIcon>
            <ListItemText primary="Profile" />
          </ListItem>
        </List>
      </Drawer>
    </>
  );
}

export default SideNav;
