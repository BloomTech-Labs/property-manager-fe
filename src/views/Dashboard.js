import React, { useCallback } from 'react';

// UI Components
import { useDispatch } from 'react-redux';
import { VertNav } from '../components/Navigation/VerticalNav';

// Add Property Form
import AddPropertyForm from '../components/AddPropertyForm/AddPropertyForm';

// Redux imports
import { createProperty } from '../store/actions';

const dispatchAddProperty = createProperty(
  'https://pt6-propman.herokuapp.com/api/properties'
);

const Dashboard = () => {
  const dispatch = useDispatch();

  const handleAddProperty = useCallback(
    ({ ...values }) => dispatch(dispatchAddProperty(values)),
    [dispatch]
  );

  return (
    <div className="dashboard">
      <VertNav />
      <h2>Dashboard</h2>
      <AddPropertyForm submit={handleAddProperty} />
    </div>
  );
};

export default Dashboard;
