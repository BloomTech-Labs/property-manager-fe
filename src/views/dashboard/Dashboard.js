/* eslint-disable react/prop-types */
/* eslint-disable react/destructuring-assignment */
import React from 'react';
// UI Components
import Container from '@material-ui/core/Container';
import { VertNav } from '../../components/Navigation/navigation';
import FloatingActions from '../../components/UI/FloatingActions';
import PropmanTheme from '../../theme/PropmanTheme';

function Dashboard(props) {
  return (
    <PropmanTheme>
      <div className="dashboard">
        <VertNav />
        <Container className="dashboard-content">{props.children}</Container>
        <FloatingActions />
      </div>
    </PropmanTheme>
  );
}

export default Dashboard;
