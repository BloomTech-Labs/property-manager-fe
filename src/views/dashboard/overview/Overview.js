import React from 'react';

export default function Overview() {
  return (
    <div className="overview">
      <h2>Overview modules are coming soon...</h2>
    </div>
  );
}
