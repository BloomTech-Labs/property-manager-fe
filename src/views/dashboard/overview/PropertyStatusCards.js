import React from 'react';

export default function PropertyStatusCards({
  propertyTotalArray,
  propertyList
}) {
  return (
    <div className="property-status-card-info">
      <div className="property-status-label">
        Occupied
        <div className="property-status-value">
          {propertyTotalArray[0] ? propertyTotalArray[0] : 0}
        </div>
      </div>
      <span> </span>
      <div className="property-status-label">
        Vacant
        <div className="property-status-value">
          {propertyTotalArray[1] ? propertyTotalArray[1] : 0}
        </div>
      </div>
      <span> </span>
      <div className="property-status-label">
        Total
        <div className="property-status-value">
          {propertyList.length > 0 ? propertyList.length : 0}
        </div>
      </div>
    </div>
  );
}
