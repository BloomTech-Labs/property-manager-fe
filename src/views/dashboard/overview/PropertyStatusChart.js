import React from 'react';
import { Doughnut } from 'react-chartjs-2';
import legendOptions from './legendOptions';

export default function PropertyStatusChart({ propertyTotalArray }) {
  const propertyStatusInfo = {
    labels: ['Occupied', 'Vacant'],
    datasets: [
      {
        data: propertyTotalArray,
        backgroundColor: ['#6c63ff', '#777'],
        hoverBackgroundColor: ['#9f99ff', '#a1a1a1']
      }
    ]
  };
  return (
    <div className="property-status-card">
      <div className="property-status-chart">
        <h4 className="property-status-label">Property Status</h4>
        <Doughnut
          data={propertyStatusInfo}
          legend={legendOptions}
          options={{ rotation: 1 * Math.PI, circumference: 1 * Math.PI }}
        />
      </div>
    </div>
  );
}
