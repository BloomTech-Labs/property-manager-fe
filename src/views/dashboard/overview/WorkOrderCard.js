import React from 'react';
import WorkOrderChart from './WorkOrderChart';

export default function WorkOrderCard({ workOrderTotalArray }) {
  return (
    <>
      {workOrderTotalArray.length > 0 ? (
        <div className="work-order-card">
          <div className="work-order-chart">
            <h4 className="work-order-label">Work Orders By Type</h4>
            <WorkOrderChart workOrderTotalArray={workOrderTotalArray} />
          </div>
        </div>
      ) : (
        <h2>No Work Orders Submitted</h2>
      )}
    </>
  );
}
