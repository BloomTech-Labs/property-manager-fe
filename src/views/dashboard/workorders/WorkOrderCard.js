/* eslint-disable react/prop-types */
import React from 'react';
import { useSelector } from 'react-redux';
import Divider from '@material-ui/core/Divider';

function WorkOrderCard({ workOrders }) {
  const properties = useSelector(state => state.propReducer.properties);

  const property = properties.filter(
    property => property.id === workOrders.unit_id
  );

  return (
    <>
      <div className="wo-card-wrapper">
        <div className="wo-card">
          <div className="card-heading">
            <h5>Workorder Unit:</h5>
          </div>
          <div className="card-content">
            <h3>{property[0].name}</h3>
          </div>
        </div>
        <Divider orientation="vertical" />
        <div className="wo-card">
          <div className="card-heading">
            <h5>Description of Issue:</h5>
          </div>
          <div className="card-content">
            <h3>{workOrders.description}</h3>
          </div>
        </div>
      </div>
    </>
  );
}

export default WorkOrderCard;
