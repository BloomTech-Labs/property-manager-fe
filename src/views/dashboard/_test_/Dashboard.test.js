import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup } from '@testing-library/react';
import { Provider } from 'react-redux';
import Container from '@material-ui/core/Container';
import initialStore from '../../../store';
import Dashboard from '../Dashboard';
import Overview from '../overview/Overview';
import PropertyStatusChart from '../overview/PropertyStatusChart';
import PropertyStatusCards from '../overview/PropertyStatusCards';
import WorkOrderCard from '../overview/WorkOrderCard';
import SideNav from '../../../components/Navigation/SideNav/SideNav';
import PropManTheme from '../../../theme/PropmanTheme';

afterEach(cleanup);

describe('With React Testing Library', () => {
  it('Shows "Hello world!"', () => {
    const { asFragment } = render(
      <Provider store={initialStore}>
        <Dashboard />
      </Provider>
    );

    expect(asFragment(<Dashboard />)).toMatchSnapshot();
  });

  it('dashboard renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Provider store={initialStore}>
        <Dashboard />
      </Provider>,
      div
    );
  });

  it('renders navigation without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Provider store={initialStore}>
        <SideNav />
      </Provider>,
      div
    );
  });

  it('renders navigation without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Provider store={initialStore}>
        <PropManTheme />
      </Provider>,
      div
    );
  });

  it('renders navigation without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Provider store={initialStore}>
        <Container />
      </Provider>,
      div
    );
  });

  it('renders "Dashboard Profile Properties Tenants Work Orders Logout" ', () => {
    const container = document.createElement('div');
    ReactDOM.render(
      <Provider store={initialStore}>
        <Dashboard />
      </Provider>,
      container
    );
    expect(container.textContent).toMatch(
      'DashboardProfilePropertiesTenantsWork OrdersLogout'
    );
  });

  it('renders "No Work Orders Or Properties Listed " when no properties or work orders created ', () => {
    const container = document.createElement('div');
    ReactDOM.render(
      <Provider store={initialStore}>
        <Overview />
        <PropertyStatusChart propertyTotalArray={[]} />
        <PropertyStatusCards propertyTotalArray={[]} propertyList={[]} />
        <WorkOrderCard workOrderTotalArray={[]} />
      </Provider>,
      container
    );
    expect(container.textContent).toMatch(
      'No Work Orders Or Properties Listed '
    );
  });

  it('renders an image', () => {
    const container = document.createElement('img');
    ReactDOM.render(
      <Provider store={initialStore}>
        <Overview />
      </Provider>,
      container
    );
    expect(container.nodeType).toBe(1);
  });

  it('renders Logout button', () => {
    const button = 'Logout';
    const { getByText } = render(
      <Provider store={initialStore}>
        <SideNav />
      </Provider>
    );
    expect(getByText(button)).toBeTruthy();
  });
});
