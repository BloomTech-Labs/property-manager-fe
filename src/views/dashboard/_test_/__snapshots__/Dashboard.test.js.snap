// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With React Testing Library Shows "Hello world!" 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-1"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <button
        aria-label="open drawer"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-6 makeStyles-drawerOpen-7"
    >
      <div
        class="MuiPaper-root MuiDrawer-paper makeStyles-drawerOpen-7 MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
      >
        <div
          class="makeStyles-toolbar-8"
        >
          <img
            alt="Freehold branding"
            class="makeStyles-logo-9"
            src="logo.png"
          />
        </div>
        <hr
          class="MuiDivider-root"
        />
        <ul
          class="MuiList-root makeStyles-list-10 MuiList-padding"
        >
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 13h8V3H3v10zm0 8h8v-6H3v6zm10 0h8V11h-8v10zm0-18v6h8V3h-8z"
                />
              </svg>
            </div>
            <div
              class="MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
              >
                Dashboard
              </span>
            </div>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
          <hr
            class="MuiDivider-root"
          />
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                />
              </svg>
            </div>
            <div
              class="MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
              >
                Profile
              </span>
            </div>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8.17 5.7L1 10.48V21h5v-8h4v8h5V10.25z"
                />
                <path
                  d="M17 7h2v2h-2z"
                  fill="none"
                />
                <path
                  d="M10 3v1.51l2 1.33L13.73 7H15v.85l2 1.34V11h2v2h-2v2h2v2h-2v4h6V3H10zm9 6h-2V7h2v2z"
                />
              </svg>
            </div>
            <div
              class="MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
              >
                Properties
              </span>
            </div>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16 11c1.66 0 2.99-1.34 2.99-3S17.66 5 16 5c-1.66 0-3 1.34-3 3s1.34 3 3 3zm-8 0c1.66 0 2.99-1.34 2.99-3S9.66 5 8 5C6.34 5 5 6.34 5 8s1.34 3 3 3zm0 2c-2.33 0-7 1.17-7 3.5V19h14v-2.5c0-2.33-4.67-3.5-7-3.5zm8 0c-.29 0-.62.02-.97.05 1.16.84 1.97 1.97 1.97 3.45V19h6v-2.5c0-2.33-4.67-3.5-7-3.5z"
                />
              </svg>
            </div>
            <div
              class="MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
              >
                Tenants
              </span>
            </div>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  clip-rule="evenodd"
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M22.7 19l-9.1-9.1c.9-2.3.4-5-1.5-6.9-2-2-5-2.4-7.4-1.3L9 6 6 9 1.6 4.7C.4 7.1.9 10.1 2.9 12.1c1.9 1.9 4.6 2.4 6.9 1.5l9.1 9.1c.4.4 1 .4 1.4 0l2.3-2.3c.5-.4.5-1.1.1-1.4z"
                />
              </svg>
            </div>
            <div
              class="MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
              >
                Work Orders
              </span>
            </div>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
          <div>
            <hr
              class="MuiDivider-root"
            />
            <div
              aria-disabled="false"
              class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemIcon-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M10.09 15.59L11.5 17l5-5-5-5-1.41 1.41L12.67 11H3v2h9.67l-2.58 2.59zM19 3H5c-1.11 0-2 .9-2 2v4h2V5h14v14H5v-4H3v4c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
              </div>
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
                >
                  Logout
                </span>
              </div>
              <span
                class="MuiTouchRipple-root"
              />
            </div>
          </div>
        </ul>
      </div>
    </div>
    <main
      class="makeStyles-content-2"
    >
      <div
        class="MuiContainer-root dashboard-content MuiContainer-maxWidthLg"
      />
    </main>
  </div>
</DocumentFragment>
`;
